### PR TITLE
feat: keyless signing

### LINKED ISSUE #2713: use keyless approach while signing and verifying GoReleaser checksums.txt file
**Is your feature request related to a problem? Please describe.**
It is not related to a problem. 

**Describe the solution you'd like**
A similar approach has been taken in cosign project[^1], thanks to @cpanato. Now, we can use this approach while signing and verifying the checksums.txt file of the GoReleaser.

Also thanks to @cpanato, he already prepared a repository[^2] to show people how they can keyless approach with GoReleaser.

**Describe alternatives you've considered**

* We should remove both `cosign.key` and `cosign.pub` files.
* We should enable cosign's experimental mode by providing the `COSIGN_EXPERIMENTAL` environment variable.
* We should output the certificate while signing the blob, and upload this certificate to the releases.
* We can use this certificate while verifying the `checksums.txt.sig` file.

**Additional context**

[^1]: https://github.com/sigstore/cosign/pull/1111
[^2]: https://github.com/caarlos0-graveyard/gorel-keyless

cc: @caarlos0 @dirien @Dentrax

### PR DESCRIPTION
**Pull request by caarlos0**
closes  #2713

depends on cosign v1.4.0

cc/ @developer-guy

### CODE REVIEW

File: www/docs/install.md [hunk 1]
```diff
@@ -142,9 +142,8 @@ You can verify it using [our public key](https://goreleaser.com/static/gorelease
       ```
     1. Verify the signature:
       ```sh
-      cosign verify-blob \
-        -key https://goreleaser.com/static/goreleaser.pub \
-        -signature checksums.txt.sig \
+      COSIGN_EXPERIMENTAL=1 cosign verify-blob \
+        --signature checksums.txt.sig \
```
Comment (developer-guy) [inline www/docs/install.md:146]: once we enable keyless mode in GoReleaser, people can use our certificate and signature while verifying without requiring experimental mode. So, the command will be like this:

```shell
$ cosign verify-blob --signature checksums.txt.sig --cert checksums.pem checksums.txt
Verified OK
```

Screen Shot 2021-12-09 at 20 27 07

WDYT @caarlos0 @dirien @dentrax?

File: www/docs/install.md [hunk 2]
```diff
@@ -160,36 +159,29 @@ You can verify it using [our public key](https://goreleaser.com/static/gorelease
...
```
Comment (developer-guy) [inline www/docs/install.md:163]: same. 😋

File: www/docs/install.md [hunk 3]
```diff
@@ -160,36 +159,29 @@ You can verify it using [our public key](https://goreleaser.com/static/gorelease
...
     1. If the signature is valid, you can then verify the SHA256 sums match with the downloaded binary:
       ```sh
       sha256sum --ignore-missing -c checksums.txt
       ```
 
-## Verifying docker images
+### docker images
 
-Our Docker image is signed with [cosign][].
+Our Docker images are signed with [cosign][].
```
Comment (developer-guy) [inline www/docs/install.md:173]: maybe we can give more information about the release, like how they can search GoReleaser binary SHA256 in the transparency log as `Tekton Chains` did its releases. WDYT?

* https://github.com/tektoncd/chains/releases/tag/v0.6.1

Comment (caarlos0) [reply www/docs/install.md:173]: Sgtm, can you add it to my branch or open a pr to it? Thankss

**Pull request merged**